### PR TITLE
Authentication flow improvements

### PR DESCRIPTION
**Pull request by lpetkov-sw**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -100,40 +98,35 @@ const CommonHome = () => {
...
+
+        userStore.setSignedInUser(filteredUser);
+        userStore.setIsLoading(false);
+      } else {
+        userStore.setSignedInUser(null);
       }
-    };
+      userStore.setIsLoading(false);
+    } catch (error) {
+      console.log('ERRROR', error);
```
Comment (jellegerbrandy) [inline App.js:124]: This is not good: we have an error, and we just log it to the console, but we leave the user in the "loading..." state without any feedback.

Either just remove the try... catch... here, or, perhaps better, show the user an error here (using haos error messaging thing)

File: App.js [hunk 2]
```diff
@@ -100,40 +98,35 @@ const CommonHome = () => {
...
-        console.log('ERRROR', error);
+        const allUserInfo = {
+          ...user._user,
+          ...appUser,
+        };
+
+        const filteredUser = filterObjectByKeys(allUserInfo, userInfoFields);
+
+        userStore.setSignedInUser(filteredUser);
+        userStore.setIsLoading(false);
```
Comment (jellegerbrandy) [inline App.js:118]: this line is not needed, we have this in l. 122

File: App.js [hunk 3]
```diff
@@ -147,7 +140,7 @@ const App = ({userStore}) => {
     };
 
     checkOnboardingStatus();
-    loadUser();
+    return subscriber; // unsubscribe on unmount
```
Comment (jellegerbrandy) [inline App.js:143]: what does "unsubscribe on unmount" mean?

File: src/Components/GSignInButton.js [hunk 4]
```diff
@@ -49,7 +48,16 @@ const GSignInButton = ({onSignIn}) => {
     }
   };
 
-  _getMnemonic = async () => {
+  _getMnemonic = async uid => {
+    // 1. Read mnemonic from the store
+    /*
```
Comment (jellegerbrandy) [inline src/Components/GSignInButton.js:53]: why comment this?

File: App.js (hunk 2 above)
Comment (lpetkov-sw) [reply App.js:118]: This line is actually removed in this PR. It's in red.

File: src/Components/GSignInButton.js (hunk 4 above)
Comment (lpetkov-sw) [reply src/Components/GSignInButton.js:53]: I have an error in calling retrieveMnemonic function. Think there is a bug with the last changes for mnemonic storing by uuid. I wrote to Hao about that.

File: App.js (hunk 1 above)
Comment (lpetkov-sw) [reply App.js:124]: This is also removed code.

File: App.js (hunk 2 above)
Comment (lpetkov-sw) [reply App.js:118]: Hm, strange on the code view, this labels are appearing in red code 

image

File: src/Components/GSignInButton.js (hunk 4 above)
Comment (jellegerbrandy) [reply src/Components/GSignInButton.js:53]: @lpetkov-sw I merged Haos PR to master, so you can rebase and then it should work

**Pull request merged**